src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -255,9 +264,14 @@ private function createFirewalls(array $config, ContainerBuilder $container)
         // add authentication providers to authentication manager
         $authenticationProviders = array_map(function ($id) {
             return new Reference($id);
-        }, array_values(array_unique($authenticationProviders)));
+        }, array_unique($authenticationProviders));
+        $authenticationManagerId = 'security.authentication.manager.provider';
+        if ($this->guardAuthenticationManagerEnabled) {
+            $authenticationManagerId = 'security.authentication.manager.guard';
+            $container->setAlias('security.authentication.manager', new Alias($authenticationManagerId));
+        }
```

Comment:
I wonder if `guardAuthenticationManagerEnabled` is true, if we should *remove* (or not load) the old system/services. Like, you're switching fully between the two systems for now. It might (?) make converting to the system easier - as once you opt into it, you are fully into it (and it's impossible to use the old system)?